localstack/config.py

```diff
@@ -870,17 +865,6 @@ def legacy_fallback(envar_name: str, default: T) -> T:
     os.environ.get("SQS_CLOUDWATCH_METRICS_REPORT_INTERVAL") or 60
 )
 
-# DEPRECATED: only applies to old lambda provider
-# Endpoint host under which LocalStack APIs are accessible from Lambda Docker containers.
-HOSTNAME_FROM_LAMBDA = os.environ.get("HOSTNAME_FROM_LAMBDA", "").strip()
```

Comment:
It is actually implemented for the new lambda provider as well in [get_main_endpoint_from_container](https://github.com/localstack/localstack/blob/469febb978283a6fea99708c8eb397c32dca6de4/localstack/services/lambda_/networking.py#L13).

Migration path: Should be replaced by `LOCALSTACK_HOST`

Reply:
As discussed in the Lambda service owner sync with @dfangl and @dominikschubert , `HOSTNAME_FROM_LAMBDA` was introduced in the new provider upon request. It is still marked as deprecated since `2.0.0` and we ultimately want users to migrate to `LOCALSTACK_HOST`

=> We continue to support the flag for now but do not advertise it. Ideally, `LOCALSTACK_HOST` should cover this scenario.